plugins/modules/purefa_pgsnap.py

```diff
@@ -670,13 +670,23 @@ def restore_pgsnapvolume(module, array):
                             overwrite=module.params["overwrite"],
                         )
                 else:
-                    res = array.post_volumes(
-                        names=[module.params["target"]],
-                        volume=VolumePost(source=Reference(name=source_volume)),
-                        with_default_protection=module.params[
-                            "with_default_protection"
-                        ],
-                    )
+                    if module.params["overwrite"]:
+                        res = array.post_volumes(
+                            names=[module.params["target"]],
+                            volume=VolumePost(source=Reference(name=source_volume)),
+                            with_default_protection=module.params[
```

Comment:
Cannot pass both variable "overwrite" & "with_default_protection", getting error as invalid parameter combination. We should restrict this in ansible play level as well to refrain users in passing both vars.